cleo_plugins/MemoryOperations/MemoryOperations.cpp

```diff
@@ -378,10 +378,15 @@ class MemoryOperations
         HMODULE ptr = nullptr;
         
         // resolve absolute path and try load
-        char buff[MAX_PATH];
-        strncpy(buff, path, sizeof(buff));
-        CLEO_ResolvePath(thread, buff, sizeof(buff));
-        ptr = LoadLibrary(buff);
+        std::string buff = path;
+        buff.resize(MAX_PATH);
+        CLEO_ResolvePath(thread, buff.data(), buff.size());
+        buff.resize(strlen(buff.data()));
+
+        // ModLoader's hooks require relative paths in LoadLibrary to work
+        FilepathRemoveParent(buff, CLEO_GetGameDirectory());
+
+        ptr = LoadLibrary(buff.c_str());
```

Comment:
why not
```
        OPCODE_READ_PARAM_FILEPATH(path);

        HMODULE ptr = LoadLibrary(path);
        if (ptr != nullptr)
        {
            m_libraries.insert(ptr);
        }
        OPCODE_WRITE_PARAM_PTR(ptr);
        OPCODE_CONDITION_RESULT(ptr != nullptr);
        return OR_CONTINUE;
```

Reply:
What about ```LoadLibrary("kernel32.dll")``` ?

Reply:
I assume the path shouldn't change and remain "kernel32.dll"?

Reply:
If you change work dir with 0A99 it will be added at the beginning then. That function did not used path reading macro on purpose.

Reply:
Wdym? If workdir is within game directory it will stripped from the path.

Reply:
If workdir is outside game dir it will be validation error. File path macro already does the same

Reply:
0A99 does verification already. Currently work dir is not guaranteed to be absolute either, so it may result in "cleo/kernel32.dll".

Reply:
So what?

Reply:
How's this a problem if it was always behaving like this??

Reply:
As it full path fails it tries again with exactly what user provided. Otherwise you wont be able to load any dll providing just the name.

Reply:
Ok, question.

```
0a99: "cleo"
Load_library "kernel32.dll"
```

```
Load_library "cleo\kernel.dll"
```


Assuming I have custom kernel32.dll in Cleo folder.
Will it be loaded in any of two variants in CLEO4 and CLEO5?

Reply:
In CLEO5 yes, in CLEO4 probably too. Load_library has some internal logic for resolving dll paths. That's why dropping hooked dlls into app directory works almost in every software.

Reply:
ok, so currently it works because we have a fallback, which ignores all workdir rules:

```
if (ptr == nullptr && !std::filesystem::path(path).has_parent_path())
```

it makes this code work
```
0a99: "cleo" 
Load_library "kernel32.dll"
```

Reply:
but why we need the change from this MR?

```
// ModLoader's hooks require relative paths in LoadLibrary to work
        FilepathRemoveParent(buff, CLEO_GetGameDirectory());
```

what issue does it solve? Doesn't CLEO_ResolvePath do it already?

Reply:
ML seems unable to install its hooks if library passed to LoadLibrary is not relative path. It happens when loading cleo plugins too.

Reply:
ML has nothing to do with virtual/absolute paths. What script was reported not working to make this change?

Reply:
Can you revert the change to 0AA2?

Reply:
> ML has nothing to do with virtual/absolute paths. What script was reported not working to make this change?

https://github.com/cleolibrary/CLEO5/blob/master/source/CScriptEngine.cpp#L682
Perhaps if changed it will become dead loop in some cases, as resolved path will still be seen as not resolved.

Reply:
If we want to make virtual path ML-compatible (i.g. relative to game dir), then we should do it in ResolvePath. stream_custom_script "cleo:\1.cs" and load_library "cleo:\1.dll" should work the same.